labgrid/strategy/ubootstrategy.py

```diff
@@ -41,19 +41,35 @@ def transition(self, status):
             self.target.activate(self.power)
             self.power.off()
         elif status == Status.uboot:
-            self.transition(Status.off)
+            if self.status != Status.off:
+                self.transition(Status.off)
```

Comment:
This is not needed. The transition method does nothing if the requested status is the same as the saved `self.status`, see above ("nothing to do"). Think of `transition()` as a state machine.

Reply:
right it's called recursively